src/plugins/intel_gpu/src/runtime/ocl/ocl_engine.cpp

```diff
@@ -54,8 +54,11 @@ ocl_engine::ocl_engine(const device::ptr dev, runtime_types runtime_type,
     _usm_helper.reset(new cl::UsmHelper(get_cl_context(), get_cl_device(), use_unified_shared_memory()));
 
 #ifdef ENABLE_ONEDNN_FOR_GPU
-    _onednn_engine = std::make_shared<dnnl::engine>(dnnl::ocl_interop::make_engine(casted->get_device().get(), casted->get_context().get()));
+    auto info = dev->get_info();
+    if (info.supports_immad == true)
+        _onednn_engine = std::make_shared<dnnl::engine>(dnnl::ocl_interop::make_engine(casted->get_device().get(), casted->get_context().get()));
```

Comment:
How will it work with impl forcing? Do we have proper checks anywhere to avoid segfault? Don't we need a possibility to test onednn integration on iGPU (in precommits)?

Reply:
We are checking that onednn engine is created properly in `get_onednn_engine`. This PR is assuming that iGPU won't use onednn. But agree with your point that it is not future-proof :(

Reply:
Have you checked if current tests cases for onednn integration work? I have a feeling that they shouldn't...

Reply:
I had a short discussion with Mingyu after checking engine creation in oneDNN.
It seems there is no logic to check cache_dir in hw initializing.
I will upload a ticket to add a way to verify cache_dir to resolve this.